core/src/main/scala/chisel3/internal/Builder.scala

```diff
@@ -178,10 +190,22 @@ private[chisel3] trait HasId extends InstanceId {
   }
 
   /** This resolves the precedence of [[autoSeed]] and [[suggestName]]
+    *
+    * @note It will become an error in the future to suggestName the same thing that autoSeed would have assigned
     *
     * @return the current calculation of a name, if it exists
     */
-  private[chisel3] def seedOpt: Option[String] = suggested_seed.orElse(auto_seed)
+  private[chisel3] def seedOpt: Option[String] = {
+    suggested_seed.zip(auto_seed).foreach {
+      case (suggested, auto) =>
+        if (suggested == auto) {
+          Builder.deprecated(
+            s"calling suggestName(${suggested}) had no effect as it is the same as the auto prefixed name, this will become an error in 3.6"
+          )
+        }
+    }
```

Comment:
```suggestion
    if (suggested_seed.isDefined && auto_seed.isDefined) {
      if (suggested_seed.get == auto_seed.get) {
        Builder.deprecated(
          s"calling suggestName(${suggested}) had no effect as it is the same as the auto prefixed name, this will become an error in 3.6"
        )
      }
    }
```
I think this is worth micro-optimizing since this method is called on every `Data`, sometimes multiple times. Your code _probably_ JIT compiles just fine but I know the above will be fast in practice (because most don't have a `suggested_seed` so it will generally be 1 load and 1 branch rather than potentially allocating a Tuple object.